src/flamegraph/mod.rs

```diff
@@ -32,23 +28,67 @@ const XPAD: usize = 10; // pad lefm and right
 const FRAMEPAD: usize = 1; // vertical padding for frames
 const PALETTE_FILE: &str = "palette.map";
 
+/// Configure the flame graph.
 #[derive(Debug, Default)]
 pub struct Options {
+    /// The color palette to use when plotting.
     pub colors: color::Palette,
+
+    /// The background color for the plot.
+    ///
+    /// If `None`, the background color will be selected based on the value of `colors`.
     pub bgcolors: Option<color::BackgroundColor>,
+
+    /// Choose names based on the hashes of function names.
+    ///
+    /// This will cause similar functions to be colored similarly.
     pub hash: bool,
+
+    /// Store the choice of color for each function so that later invocations use the same colors.
+    ///
+    /// With this option enabled, a file called `palette.map` will be created the first time a
+    /// flame graph is generated, and the (random) color chosen for each function will be written
+    /// into it. On subsequent invocations, functions that already have a color registered in that
+    /// file will be given the stored color rather than be assigned a new one. New functions will
+    /// have their colors persisted for future runs.
+    ///
+    /// This feature was first implemented [by Shawn
+    /// Sterling](https://github.com/brendangregg/FlameGraph/pull/25).
     pub consistent_palette: bool,
+
+    /// Assign extra attributes to particular functions.
+    ///
+    /// In particular, if a function appears in the given map, it will have extra attributes set in
+    /// the resulting SVG based on its value in the map.
     pub func_frameattrs: FuncFrameAttrsMap,
+
+    /// Whether to plot a plot that grows top-to-bottom or bottom-up (the default).
     pub direction: Direction,
+
+    /// The title for the flame chart.
     pub title: String,
 }
 
+/// The direction the plot should grow.
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub enum Direction {
+    /// Stacks grow from the bottom to the top.
+    ///
+    /// `main` will be at the bottom.
```

Comment:
`main` might be too language specific, and even for language with `main`, it is possible to have a stack with no `main` (launching a `Thread` in java for example). Maybe change to `entry point` ?